.changeset/link-add-underline.md

```diff
@@ -0,0 +1,10 @@
+---
+"@salt-ds/core": minor
+---
+
+Added `underline` prop to `Link`. It controls when underline text decoration is applied to a link.
+
+```
+<Link underline="link">Underline link</Link>
```

Comment:
NIT: It controls when underline text decoration is applied to a link.

From a consumer's POV, "hover" is a state and explains itself but "link" is not, so might need some clarification (all links are links, so when does is it applied?)